embc-app/ClientApp/src/app/volunteer-list/volunteer-list.component.ts

```diff
@@ -19,194 +18,80 @@ export class VolunteerListComponent implements OnInit {
-  pageSize: number; // how many entries are on the page
-  previousQuery: string; // a place to save the last query parameters
-  sort: string = ''; // how do we sort the list
-  collectionSize: number = 0; // how large is the collection?
-  maxSize = 20; // how many results should the UI show?
-  boundaryLinks = true; // do we show the jump to first and last page links?
-
-  // the search form and associated toggles (show all, show only admins, show only regular users)
-  form: FormGroup;
-  // this is the correct path prefix for the user
+  defaultSearchQuery: VolunteerSearchQueryParameters = {
+    offset: 0,
+    limit: 20
+  };
+  queryString: string;
+  previousQuery: SearchQueryParameters = {}; // a place to save the last query parameters
+  sort = ''; // how do we sort the list
+
+  // this is the correct path prefix for the user routing
   path: string;
 
   constructor(
     private router: Router,
-    private route: ActivatedRoute,
-    private fb: FormBuilder,
-    private volunteers: VolunteerService,
-    private organizations: OrganizationService,
+    private volunteerService: VolunteerService,
     private authService: AuthService,
     private uniqueKeyService: UniqueKeyService
   ) { }
 
-  // convenience getters
-  get results(): Volunteer[] {
-    // if there are non null results in the global variable to contain the volunteers return the collection of data otherwise return null
-    return this.resultsAndPagination ? this.resultsAndPagination.data : null;
-  }
-
-  get pagination(): PaginationSummary {
-    // if there are non null results in the global pagination summary to contain the results and pagination return the metadata otherwise return null
-    return this.resultsAndPagination ? this.resultsAndPagination.metadata : null;
-  }
-
   ngOnInit() {
+    // save the path for routing later
     this.authService.path.subscribe(p => this.path = p);
-    // initialize the form component
-    this.initSearchForm();
     // initialize the global variables. When done get volunteers associated with the organization
-    this.initVars()
-      .subscribe(org => {
-        // save into the global state
-        if (org) {
-          this.currentOrganization.id = org.id;
-        }
-        // fetch volunteers now that we may know the org
-        this.getVolunteers();
+    this.authService.getCurrentUser()
+      .subscribe(u => {
+        // collect the current user
+        this.volunteerService.getVolunteerById(u.contactid)
+          .subscribe(
+            v => {
+              // save the organization ID for future search queries
+              this.defaultSearchQuery.org_id = v.organization.id;
+              // get the results
+              this.getVolunteers(this.defaultSearchQuery)
+                .subscribe(r => {
+                  this.resultsAndPagination = r;
+                });
+            },
+            err => {
+              this.notFoundMessage = err;
+            }
+          );
       });
```

Comment:
I recommend adding handling for getCurrentUser() error here, ie in the subscrive you handle 'u' but not the error case.